src/Entities/Menu.java

```diff
@@ -1,52 +1,107 @@
 
 import java.util.HashMap;
-import java.util.List;
 
 public class Menu {
-    private HashMap<IFood, ISingleton> menu;
 
-    public Menu() {
-        this.menu = new HashMap<>();
+    private HashMap<IFood, Object[]> menu;
+
+
+    /**
+     * A method for adding a new food item to a menu
+     *
+     * @param food the new food item
+     * @param price the price for the food item
+     * @param availability the availability of the food item
+     */
+    public void addFood(IFood food, float price, boolean availability){
+        Object[] foodInfo = {price, availability};
+        this.menu.put(food, foodInfo);
     }
 
-    public Menu(HashMap<IFood, ISingleton> menu){
-        this.menu = menu;
+    /**
+     * A method for removing a food item to a menu
+     *
+     * @param food the new food item
+     */
+    public void removeFood(IFood food) {
+        this.menu.remove(food);
     }
-    public void MenuItem(IFood[] foods, ISingleton[] singleton) {
-        for (int i = 0; i < foods.length; i++) {
-            menu.put(foods[i], singleton[i]);
-        }
+
+    /**
+     * A method for modifying a food item to a menu
+     *
+     * @param food the  food item
+     * @param price the new price for the food item
+     * @param availability the updated availability of the food item
+     */
+    public void setFood(IFood food, float price, boolean availability){
+        Object[] foodInfo = this.menu.get(food);
+        foodInfo[0] = price;
+        foodInfo[1] = availability;
     }
 
-    public HashMap<IFood, ISingleton> getMenuContents() {
+
+
+    /**
+     * A method for getting the food items in a menu
+     * @return the items of the menu in  hashmap
+     */
+    public HashMap<IFood, Object[]> getContents() {
         return menu;
     }
 
+    /**
+     * A method for getting the price of a food item
+     * @param item the food item we want the price of
+     * @return the price of the food item
+     */
     public float getPrice(IFood item) {
-        if (menu.containsKey(item)) {
-            return menu.get(item).getPrice();
+        // if price is -1, then calculate price from the components
+        if (this.menu.containsKey(item)) {
+            return (float) this.menu.get(item)[0];
         } else
             return 0;
     }
 
+    /**
+     *A method for getting the components of a food item if it has any
+     * @param item the food item we want the description
+     * @return the description of the food item as a string
+     */
     public String getDescription(IFood item) {
         if (menu.containsKey(item)) {
-            return menu.get(item).getDescription();
-        } else
-            return null;
-    }
-    public void setMenu(HashMap<IFood, ISingleton> menu){
-        this.menu = menu;
+            StringBuilder description = new StringBuilder();
+            for(ISingleton component: item.getComponents()){
+                description.append(component.getDescription());
```

Comment:
I thought we removed descriptions from singletons?

Reply:
I thought we still needed them to build out the description for food. I think that it's to help with nesting, so that food wouldn't have to call addOn.getDescription too.